File: src/main/java/sunshine/infrastructure/restclient/openmeteo/OpenMeteoResponse.java
package sunshine.infrastructure.restclient.openmeteo;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * OpenMeteo API 응답을 매핑하는 DTO.
 */
public record OpenMeteoResponse(Current current) {

  public record Current(
      @JsonProperty("temperature_2m")
      double temperature2m,
      @JsonProperty("relative_humidity_2m")
      int relativeHumidity2m,
      @JsonProperty("apparent_temperature")
      double apparentTemperature,
      @JsonProperty("weather_code")
      int weatherCode,
      @JsonProperty("wind_speed_10m")
      double windSpeed10m
  ) {
  }
}

File: src/main/kotlin/com/sunshine/infrastructure/adapter/OpenMeteoResponse.kt
package com.sunshine.infrastructure.adapter

data class OpenMeteoResponse(
    val current: OpenMeteoCurrent?
)

data class OpenMeteoCurrent(
    val time: String?,
    val temperature_2m: Double?,
    val apparent_temperature: Double?,
    val wind_speed_10m: Double?,
    val weather_code: Int?,
    val relative_humidity_2m: Int?
)
